components/encoder/fake/encoder.go

```diff
@@ -28,11 +27,7 @@ func init() {
 			cfg config.Component,
 			logger golog.Logger,
 		) (interface{}, error) {
-			e := &Encoder{}
-			e.updateRate = cfg.ConvertedAttributes.(*AttrConfig).UpdateRate
-
-			e.Start(ctx)
-			return e, nil
+			return NewFakeEncoder(ctx, cfg)
```

Comment:
[opt] I prefer private creator functions,  unless this is actually being called from somewhere else.
```suggestion
			return newFakeEncoder(ctx, cfg)
```